src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,121 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final StorageDaoImpl storageDao = new StorageDaoImpl();
```

Comment:
Directly using `storageDao.add()` in the `setUp` method bypasses the registration logic in `RegistrationServiceImpl`. Consider using `registrationService.register()` to ensure all validation checks are properly tested, as mentioned in the previous review.